manifests/install.pp

```diff
@@ -50,20 +53,37 @@
   }
 
   if $::osfamily == 'windows' {
-    if $::puppet_agent::is_pe == true and empty($::puppet_agent::source) and defined(File["${::puppet_agent::params::local_packages_dir}/${package_file_name}"]) {
-      class { 'puppet_agent::windows::install':
-        package_file_name => $package_file_name,
-        source            => windows_native_path("${::puppet_agent::params::local_packages_dir}/${package_file_name}"),
-      }
-    } else {
-      class { 'puppet_agent::windows::install':
-        package_file_name => $package_file_name,
-        source            => $::puppet_agent::source,
+    # Prevent re-running the batch install
+    if versioncmp("${::agent_version}", "${package_version}") < 0 {
```

Comment:
So this blocks downgrades. That seems ok for now, since we don't officially support them.